bin/upversion

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash -e
+#
+# Usage:
+#   bin/upversion NEW_VERSION
+# 
+# Changes the version where it is hardcoded in UDMI files, performing checks
+# that the new version is available, and that the files can be safely updated.
+# Modifications:
+#   - schema/*.json (all files)
+#   - etc/categories.json 
+#   - tests/**/*.json and validator/traces/**/*.out - 
+#     all files must be added to etc/upversion.txt only files preceeded
+#     with a 'y' are updated. Only works with "pretty"/non-minified JSON
+#   - documentation in-line exammples
+#   - specific java files (Pubber.java, ConfigUtil.java, LocalDevice.java)
+
+UPVERSION_LIST=etc/upversion.txt
+TESTS_REGEX="^(\s{0,4}\"version\"\s*:\s*)([0-9.\"]*)(,?)"
```

Comment:
can you use jq instead of this?  Seems very unstable... e.g. you can set a value with something like (from elsewhere in scripts):

bin/reset_config:    jq .system.testing.sequence_name=\"${config_file%.json}\" > ${dst_config}

Reply:
Tried `jq` but opted for sed because:
- For files in test/*,  `jq` does not support comments, so cannot be used for these files.
- For files in schema/* `jq` 
    - adding a field puts it at the bottom, whereas it makes more sense for this to be at the top
    - reformats (although very little) content, namely enforces one item per line in lists.
For all changes, there are checks that there is one of whatever it is expecting to replace for sanity